### PR TITLE
docs revisions

### PR DESCRIPTION
**Pull request by petyaracz**
Following up on some great work on the swarm docs. More emphasis on non-developers as target audience. 

Edits summary (for conv. sake):
Changes mostly affect sections 2,3,4,5 (:5.2,5.3,5.4,5.5).
    General: tidied up links, fixed some spelling mistakes.
    Detailed list of changes:
    s1 added a note on CLI usage, inspired by the old Ethereum Frontier docs
    s2 tidied up cross-platform installation, added tabs, added docker
    install option
    s3 tidied up cross-platform interaction, added tabs, added some
    clarification text, added a developers subsection to end
    s4 tidied up examples, added reference table (to some extent duplicates
    but also expands on Configuration section), added cli tutorial to end
    s5.2 fronted cli section, added examples
    s5.3 added wget note
    s5.4-5 added/tidied up examples, added cli tutorial to end
    s6 added note about pss and cli.

### CODE REVIEW

**Comment by justelad** [comment]
is it possible to get a rebase? or the amounts of conflicts is really high?

**Comment by petyaracz** [comment]
I don't know why I brought all my dummy commits with, I rebased them and can't see them locally X(.
I can rebase to upstream/master, I don't think there are conflicts, I kept my clone up-to-date. Should I go ahead?

**Comment by petyaracz** [comment]
lemme add a docker section and then i'll rebase

**Comment by zelig** [comment]
awesome work @petyaracz 
eagerly waiting for the rebase and docker section. for the latter see https://github.com/ethersphere/swarm-guide/pull/127

File: contents/usage/act.rst [hunk 1]
```diff
@@ -156,3 +168,100 @@ When accessing a resource protected by EC keys, the node that requests the conte
...
+Access control in the CLI: example usage
+-----------------------------------------
+
+.. tabs::
+
+  .. group-tab:: Passwords
+
+    First, we create a simple test file. We upload it to swarm.
+    
+    .. code-block:: none
```
Comment (zelig) [inline contents/usage/act.rst]: all these examples the file we upload should be encrypted

File: contents/architecture.rst [hunk 2]
```diff
@@ -265,8 +265,7 @@ This recursive process of constructing the Swarm hash tree will result in a sing
...
 
-Swarm exposes the FileStore API via the :ref:`bzz-raw` URL scheme directly on the http local proxy server (see :ref:`BZZ URL Schemes` and :ref:`API Reference`). This API allows file upload via POST request as well as file download with GET request. Since on this level the files have no mime-type associated, in order to properly display or serve to an application, the ``content_type`` query parameter can be added to the url. This will set the proper content type in the HTTP response.
-
+Swarm exposes the FileStore API via the :ref:`bzz-raw` URL scheme directly on the http local proxy server (see :ref:`bzz protocol suite` and :ref:`API Reference`). This API allows file upload via POST request as well as file download with GET request. Since on this level the files have no mime-type associated, in order to properly display or serve to an application, the ``content_type`` query parameter can be added to the url. This will set the proper content type in the HTTP response.
```
[the remark this answers is not in the document]
Comment (petyaracz) [reply contents/architecture.rst]: fixed

File: contents/architecture.rst [hunk 3]
```diff
@@ -275,10 +274,10 @@ The Swarm :dfn:`manifest` is a structure that defines a mapping between arbitrar
 Manifests provide the main mechanism to allow URL based addressing in Swarm. The domain part of the URL maps onto a manifest in which the path part of the URL is looked up to arrive at a file entry to serve.
 
 Manifests are currently respresented as a compacted trie (http://en.wikipedia.org/wiki/Trie) , with individual trie nodes serialised as json. The json structure has an array of :dfn:`manifest entries` minimally with a path and a reference (Swarm hash address). The path part is used for matching the URL path, the reference may point to an embedded manifest if the path is a common prefix of more than one path in the collection.
-When you retrieve a file by url, Swarm resolves the domain to a reference to a root manifest, which is recursively traversed to find the matchin path (see :ref:`Manifests`).
+When you retrieve a file by url, Swarm resolves the domain to a reference to a root manifest, which is recursively traversed to find the matchin path.
```
[the remark this answers is not in the document]
Comment (petyaracz) [reply contents/architecture.rst]: fixed

File: contents/architecture.rst [hunk 4]
```diff
@@ -275,10 +274,10 @@ The Swarm :dfn:`manifest` is a structure that defines a mapping between arbitrar
...
 
 The high level API to the manifests provides functionality to upload and download individual documents as files, collections (manifests) as directories. It also provides an interface to add documents to a collection on a path, delete a document from a collection. Note that deletion here only means that a new manifest is created in which the path in question is missing. There is no other notion of deletion in the Swarm.
-Swarm exposes the manifest API via the `bzz` URL scheme, see :ref:`BZZ URL Schemes`.
+Swarm exposes the manifest API via the `bzz` URL scheme, see :ref:`bzz protocol suite`.
```
[the remark this answers is not in the document]
Comment (petyaracz) [reply contents/architecture.rst]: fixed

File: contents/installation.rst [hunk 5]
```diff
@@ -7,35 +7,23 @@ Swarm is part of the Ethereum stack, the reference implementation is currently a
 
 Swarm runs on all major platforms (Linux, macOS, Windows, Raspberry Pi, Android, iOS).
 
+Swarm was written in golang and requires the go-ethereum client **geth** to run.
```
[the remark this answers is not in the document]
Comment (petyaracz) [reply contents/installation.rst]: removed line

**Comment by justelad** [comment]
@petyaracz no rebase?

**Comment by petyaracz** [comment]
working on it, got [stuck](https://stackoverflow.com/questions/53356781/git-rebase-doesnt-move-past-file-deleted-by-us/53359618#53359618).

**Comment by petyaracz** [comment]
the offender is contents/usage/mru.rst

**Comment by justelad** [comment]
i just tried it myself.
the problem is that `mru.rst` got renamed to `feed.rst`, and you've probably created a merge commit on your side at some point after keeping on adding changes to `mru.rst` and that screwed things up a bit (i'm guessing it was the conflicts that originally showed up in the PR). it would be fine to merge but its a pain to review.

in any case this is possible to deal with, if you know which changes you did to that file, just `git add` the `mru.rst` file, finish the rebase, then just manually add your changes to `feed.rst`, or if there were not changes in between - you could create a patch from the relevant commits, then `git rm mru.rst`

**Comment by petyaracz** [comment]
Thanks. I went on with keeping mru.rst in the meantime and the rebase did continue, but loads of other conflicts popped up. maybe a merge is easier?

**Comment by petyaracz** [comment]
trying from scratch

File: contents/usage/act.rst [hunk 6]
```diff
@@ -3,14 +3,20 @@ Access Control
...
     - For an undisclosed party - where only one grantee can access the content ``(pk)``
 
     - For a number of undisclosed parties - where every grantee can access the content ``(act)``
 
+**Creating** access control for content is currently supported only through CLI usage.
+
+**Accessing** restricted content is available through CLI and HTTP. When accessing content which is restricted by a password `HTTP Basic access authentication <https://en.wikipedia.org/wiki/Basic_access_authentication>`_ can be used out-of-the-box.
+
+.. important:: When accessing content which is restricted to certain EC keys - the node which exposes the HTTP proxy that is queried must be started with the granted private key as its ``bzzaccount`` CLI parameter.
+
```
[the remark this answers is not in the document]
Comment (petyaracz) [reply contents/usage/act.rst]: added a warning in a box

File: contents/usage/act.rst (hunk 1 above)
Comment (petyaracz) [reply contents/usage/act.rst]: added.

**Pull request closed without merging**